t/GetShortPathName.t

```diff
@@ -12,7 +12,7 @@ Win32::CreateFile($path);
 ok(-f $path);
 
 my $short = Win32::GetShortPathName($path);
-ok($short, qr/^\S{1,8}(\.\S{1,3})?$/);
+ok(length $short);
```

Comment:
This completely disables the test, no? Because `length` will always be greater than 0.

I like how @richardleach's modification tries to skip it only when it is not supported, but I'm not sure that the short name is always `LONGPA~1`. If it were, then we wouldn't need the regular expression but could do a direct string comparison.

Reply:
Short name creation is deterministic and we're testing in a freshly-created folder, so I think this should work.

Reply:
Although for consistency, it should be the same check in GetShortPathName.t and in Unicode.t. @wchristian's method in Unicode.t doesn't need a new file to be created, so is tidier in that regard.